addons/website_event/models/website_visitor.py

```diff
@@ -76,6 +77,11 @@ def _search_event_registered_ids(self, operator, operand):
 
         return [('id', 'in', visitor_ids)]
 
+    def _inactive_visitors_domain(self):
+        """ Visitors registered to events are considered always active and should not be deleted. """
+        domain = super(WebsiteVisitor, self)._inactive_visitors_domain()
+        return expression.AND([domain, [('event_registration_ids', '=', False)]])
```

Comment:
Also to check. I suppose this will be equivalent to "ID, in, ..." .

Side remark: instead of SQL (yeah, my comments are progressive), maybe sub-searches ? id, in, _search... ?

Reply:
To make sure I understand, you suggest each sub-modules makes its own sub-search and then complete the domain with "and ID in (x, y, z)" ?

I seem to recall the parameter list is limited (to something like 2000 parameters).
Since the website.visitor table grows fast, I think it's possible that we would have more than 2000 ids to specify.

If we want to go that route, then I will have to rework the code to build a list of IDs and then run several unlink calls grouped by at most 2000 IDs.

At that point I'm not sure it's more efficient / readable, any thoughts?

Reply:
2many domains are evaluated and replaced by "id in ...", which means potentially all visitors (see sub comment). I propose to discuss only on the lead part as discussion will be the same :D .

Reply:
Solved.